logprep/processor/generic_adder/mysql_connector.py

```diff
@@ -0,0 +1,140 @@
+"""This module is used to connect to a MySQL database and to retrieve data from a SQL table."""
+
+import time
+from logging import Logger
+
+import mysql.connector as db
+
+
+class MySQLConnector:
+    """Used to connect to a MySQL database and to retrieve data from a table if it has changed."""
+
+    def __init__(self, sql_config: dict, logger: Logger):
+        """Initialize the MySQLConnector.
+
+        Parameters
+        ----------
+        sql_config : dict
+           SQL configuration dictionary.
+        logger : logging.Logger
+           Logger to use.
+
+        Returns
+        -------
+        bool
+            True if the SQL table has changed, False otherwise.
+
+        """
+        self._logger = logger
+
+        self.con = db.connect(
+            user=sql_config["user"],
+            password=sql_config["password"],
+            host=sql_config["host"],
+            database=sql_config["database"],
+            port=sql_config.get("port", 3306),
+        )
+
+        self.cur = self.con.cursor()
+
+        self.target_column = sql_config["target_column"]
+        self._add_target_column = sql_config.get("add_target_column", False)
+
+        self.table_name = sql_config["table"]
+
+        self._timer = sql_config.get("timer", 60 * 3)
+        self._last_check = 0
+        self._last_table_checksum = None
+
+    def check_change(self) -> bool:
```

Comment:
Ah seems like I'm in the renaming mood today :D This one I would call `has_changed`. It returns a boolean so later when this function is used it is nice to just read `if connector.has_changed()`.

```suggestion
    def has_changed(self) -> bool:
```